snooty/language_server.py

```diff
@@ -429,7 +429,16 @@ def m_text_document__resolve(
             )
             return str(resolved_target_path)
         elif resolveType == "directive":
-            return str(self.project.config.source_path) + fileName
+            """If the filename has a .rst extension, it might be converted from
+            a YAML file. We want to get its original file path in that case."""
+            if ".rst" in fileName:
+                filePath = str(self.project.config.source_path) + fileName
```

Comment:
Python variable names should be `snake_case`. I know this file is a little inconsistent about that

I think we also want `file_path = self.project.config.source_path / fileName` so the path manipulation happens on path types rather than as string manipulation